sim/src/index/entity.test.cpp

```diff
@@ -0,0 +1,73 @@
+#include <gtest/gtest.h>
+#include <core/ics/componentSet.h>
+#include <index/entity.h>
+
+#define TEST_SUITE EntityIndex
+
+using namespace ics::index;
+
+TEST(TEST_SUITE, AddAndRemoveCompStoreId) {
+    Entity index;
+    auto entityId = index.addEntityId();
+
+    ASSERT_EQ(index.getComponents(entityId).size(), 0);
+    index.addComponent(entityId, std::pair(1, 1));
+    index.addComponent(entityId, std::pair(2, 2));
+    index.addComponent(entityId, std::pair(1, 2));
+    ASSERT_EQ(index.getComponents(entityId).size(), 3);
+    index.removeComponent(entityId, std::pair(1, 1));
+    index.removeComponent(entityId, std::pair(2, 2));
+    ASSERT_EQ(index.getComponents(entityId).size(), 1);
+
+    auto firstElement = *index.getComponents(entityId).begin();
+    ASSERT_EQ(firstElement.first, 1);
+    ASSERT_EQ(firstElement.second, 2);
+}
+
+TEST(TEST_SUITE, AddMultipleEntities) {
+    Entity index;
+    auto entityId1 = index.addEntityId();
+    auto entityId2 = index.addEntityId();
+
+    ASSERT_NE(entityId1, entityId2);
+
+    index.addComponent(entityId1, std::pair(1, 1));
+    index.addComponent(entityId2, std::pair(3, 4));
+    index.addComponent(entityId2, std::pair(3, 8));
+
+    ASSERT_EQ(index.getComponents(entityId1).size(), 1);
+    ASSERT_EQ(index.getComponents(entityId2).size(), 2);
+}
+
+TEST(TEST_SUITE, UseIndexIdFilter) {
+    Entity index;
+    auto entityId = index.addEntityId();
+    auto filter = index.id(entityId);
```

Comment:
Even with the test its really hard to follow what is going on with this `.id()` method. 
After digging abit, I think this means that of the component store Ids that is attached to the entity and return the ids.